theme/layouts/_navigation.ejs

```diff
@@ -1,9 +1,40 @@
-<nav id="main-navigation">
+<% const headerMenuItems = ["documentation", "tutorial", "ready-projects", "what's-new"];
+const regularExpression = /\\|\//;
+const flyoutPages = site.queryPages(({originalPathname}) => headerMenuItems.includes(originalPathname.split(regularExpression)[0]));
+
+let navigation = [];
+let obj = {};
+let childFirst = {};
+let childSecond = {};
+
+for (const {originalPathname, pathname, title} of flyoutPages) {
+orPath0 = originalPathname.split('\\')[0];
+orPath1 = originalPathname.split('\\')[1]
+orPath2 = originalPathname.split('\\')[2]
+
+  if (orPath1 && orPath2) {
+    childSecond = {title, pathname};
+    childFirst = {'title':orPath1, 'pathname':`${orPath0}\\${orPath1}`, 'child':childSecond};
+    obj = {'title':orPath0, 'pathname':orPath0, 'child':childFirst};
+    navigation.push(obj);
+  } else if (orPath1 && !orPath2) {
+        childFirst = {title, pathname};
+        obj = {'title':orPath0, 'pathname':orPath0, 'child':childFirst};
+        navigation.push(obj);
+    } else if (!orPath1 && !orPath2) {
+      obj = {'title':orPath0, 'pathname':orPath0};
+        navigation.push(obj);
+    }
+};
+console.log(navigation)
+%>
```

Comment:
In general you can simplify your task here, looking into your implementation I see that you wish to construct the connection between folder structures using originalPathname which is relative path to the file from `pages` directory(minus index files and file extensions).

The only relevant information for constructing the connection you have here is the top level menu items - `["documentation", "tutorial", "ready-projects", "what's-new"];`.

So your first task is to find out who are the direct children of each of that items, by direct I don't refer to "grand children"(humanly speaking). So let's understand what we understand as direct children, example if you have a structure:
```
documentation
  actions
    inject
    bg-inject
  tutorials
...
```

For documentation direct children are `actions` and `tutorials`, so now you would like to construct a query that would result into those, but not into `inject` or `bg-inject`. As you have this original paths:
```
documentation
documentation/actions
documentation/actions/inject
documentation/actions/bg-inject
documentation/tutorials
```

Now you need to create a query that will get pages for `documentation/actions` and `documentation/tutorials` originalPathname. In order to make that query works you need to ensure that first part of the URL is `documentation` it has "second part" and missing third part. so basically if we were about to use `originalPathname.split("/")`, then we need to ensure that `[0]` is `documentation`, `[1]` exist and `[2]` is missing.

```
for (topLevelItem of topLevelItems) {
  const secondLevel = site.queryPages(({originalPathname}) => originalPathname.startsWith(`${topLevelItem}/`) && originalPathname.split("/").length === 2);
}  
```

or 

```
for (topLevelItem of topLevelItems) {
  const secondLevel = site.queryPages(({originalPathname}) => {
    const items = originalPathname.split("/");
    return (items[0] === topLevelItem && items.length === 2)
  });
}  
```

Then you can construct same logic for 3-rd level items, please note you don't have to hardcode `items.length === 2)` it was for demonstration purposes.

You can also use recursion to nest as much level you need.